server/src/resolvers/mutation.ts

```diff
@@ -66,6 +66,44 @@ const Mutation = {
       message: "Your account successfully created.We've send you a mail for account verification",
     };
   },
+  async verifyUserEmail(parent, args, { prisma, request }: contextType, info): Promise<{ success: boolean }> {
+    const { username, token } = args;
+
+    try {
+      // check for username existence
+      const user = await prisma.user.findFirst({
+        where: {
+          username,
+        },
+      });
+
+      if (!user) {
+        return { success: false };
+      }
+
+      /*
+        check for verificationCode from db if it's match
+        with request set it as empty
+      */
+
+      if (token === user.verificationCode) {
+        await prisma.user.update({
+          where: {
+            username,
+          },
+          data: {
+            verificationCode: null,
+            isActive: true,
+          },
+        });
+        return { success: true };
+      }
+
+      return { success: false };
+    } catch (err) {
+      throw new Error("Server Error");
```

Comment:
would suggest to throw more appropriate/better readable errors like
user not found
authentication error if token did not verify, etc.

Reply:
Error are mostly unexpected one in this case. 

Example: DB is not responding or something wrong in backend logic.